cpp/tests/utilities/column_wrapper.hpp

```diff
@@ -477,5 +480,219 @@ class strings_column_wrapper : public detail::column_wrapper {
   }
 };
 
+/**
+ * @brief `column_wrapper` derived class for wrapping columns of lists.
+ */
+class lists_column_wrapper : public detail::column_wrapper {
+ public:
+  /**
+   * @brief Construct a lists column containing a single list of fixed-width
+   * type from an initializer list of values.
+   *
+   * Example:
+   * @code{.cpp}
+   * Creates a LIST column with 1 list composed of 2 total integers
+   * [{0, 1}]
+   * lists_column_wrapper l{0, 1};
+   * @endcode
+   *
+   * @param elements The list of elements
+   */
+  template <typename T, std::enable_if_t<is_fixed_width<T>()>* = nullptr>
+  lists_column_wrapper(std::initializer_list<T> elements) : column_wrapper{}
```

Comment:
So the issue here is that this style of initialization doesn't work with the single fixed_width_column_wrapper constructor.

`list_column_wrapper<int32_t> list( {1, 2, 3, 4} );

Compiler doesn't seem to be able to figure out what you're intending.  You have to explicitly invoke

`list_column_wrapper<int32_t> list( fixed_width_column_wrapper<int32_t>({1,2,3,4}) )`

I'm not really a huge fan of this because it makes reading declarations of these things much harder to read.   As it is right now, they can be pretty eye-watering:

```
test::lists_column_wrapper list{{{{1, 2}, {3, 4}}, {{5, 6, 7}, {0}}},
                                    {{{-1, -2}, {-3, -4}}, {{-5, -6, -7}, {0}}}};
```

Injecting fixed_width_column_wrapper all over the place in there would really obliterate things.  You can semi-dodge that with something like:

```
using L = cudf::test::fixed_width_column_wrapper<int>
test::lists_column_wrapper list{{{L{1, 2}, L{3, 4}}, {L{5, 6, 7}, L{0}}},
                                    {{L{-1, -2}, L{-3, -4}}, {L{-5, -6, -7}, L{0}}}};
```

But it's less clean.  I feel like erring on the side of "cleanest/most-readable use for the end user" is preferable to a smaller amount of code in the wrapper.

Reply:
Additionally, if you go with the template method,  the compiler interprets this:

`cudf::test::lists_column_wrapper<int> list3{ {}, {} };`

as

`cudf::test::lists_column_wrapper<int> list3{ 0, 0 };`